### PR TITLE
Pull in Edwin's scripts

### PR DESCRIPTION
**Pull request by patrick-rivos**

### CODE REVIEW

File: scripts/compare_all_artifacts.py [hunk 1]
```diff
@@ -67,8 +67,8 @@ def compare_all_artifacts(current_hash: str):
             except (RuntimeError, ValueError) as err:
```
Comment (ewlu) [inline scripts/compare_all_artifacts.py:67]: Should be updating compare_logs to include the `current_hash_committed` arg

File: scripts/compare_testsuite_log.py [hunk 2]
```diff
@@ -316,7 +324,7 @@ def compare_testsuite_log(previous_log_path: str, current_log_path: str):
     return classified_gcc_failures
 
 
-def gccfailure_to_summary(failure: Dict[LibName, GccFailure], failure_name: str, previous_hash: str, current_hash: str):
+def gccfailure_to_summary(failure: Dict[LibName, GccFailure], failure_name: str, previous_hash: str, current_hash: str, current_hash_committed: bool):
```
Comment (ewlu) [inline scripts/compare_testsuite_log.py:327]: Is this for precommit if the patch is already merged? If so, how do we determine that?

**Review by ewlu — CHANGES_REQUESTED** [review]
Do we currently have something to determine if a patch is already merged?

File: scripts/compare_testsuite_log.py (hunk 2 above)
Comment (patrick-rivos) [reply scripts/compare_testsuite_log.py:327]: gcc-postcommit-ci will set that to yes, precommit will conservatively say it's not already committed

**Review by ewlu — APPROVED** [review]

**Pull request merged**